### DISCUSSION TITLE
2 Relay system with 4 inputs

### DISCUSSION THREAD

**Discussion by guydvir2** [arendst/Tasmota#11332] [Q&A] [answered]
Hi,
Just to be sure I'm doing it the right way ( since it works so far ):
1) I have a Sonoff Dual ( and another version with Node-MCU and dual Relay module ), that control a window blinds ( roller ).
2) It was adjusted using almost every step in the DOCs as a shutter and a 3 state rocker switch that uses `GPIO3` and `GPIO9` to control UP/OFF/DOWN.
3) Up till here, every thing works as expected. 
4) Now I have a garden window which we enter home from drive way straight into the living room. This window has a keypad on the outside that with the appropiate code it opens and closes that window. So Actually it is a window with 2 switches (UP/OFF/DOWN).
5) So if I get back to my other window design, `Switch1` controls `Relay1` (up), `Switch2` controls `Relay2` (down), and now I have `Switch3` and `Switch4` that have to control `Relay1` and `Relay2` respectively. 
6) Assigning `D4` and `D5` on Node-MCU as `Switch1` and `Switch2` again didn't make the trick ( now I can say "of course" ), so I added the following `Rule3` :

`rule3 ON switch3#state DO POWER1 %value% endon`
`rule3 +ON switch4#state DO POWER2 %value% endon`

7) For now it works, but is it the RIGHT way? or perhaps some bugs will follow ? 

Guy

**Comment by sfromis** [comment, marked as answer]
Assuming that it works as you want, including combinations with the "primary" buttons, I see nothing "wrong" in what you do there. As usual, there is not one single "right" way, freedom of will and all that ;-)

**Reply by guydvir2** [reply]
Well I was more concern about doing something wrong or perhaps less correct.
10X
